cli/azd/pkg/tools/azcli/container_registry.go

```diff
@@ -68,12 +84,56 @@ func (crs *containerRegistryService) GetContainerRegistries(
 	return results, nil
 }
 
+func (crs *containerRegistryService) Login(ctx context.Context, subscriptionId string, loginServer string) error {
+	// First attempt to get ACR credentials from the logged in user
+	dockerCreds, tokenErr := crs.getTokenCredentials(ctx, subscriptionId, loginServer)
+	if tokenErr != nil {
+		// If that fails, attempt to get ACR credentials from the admin user
+		adminCreds, adminErr := crs.getAdminUserCredentials(ctx, subscriptionId, loginServer)
```

Comment:
Is this the correct behavior long-term? Or is this just added for backcompat, and needs revisiting later?

Reply:
FWIW, I think it would be nicer long term if we could just remove it.  It would be good for us to understand under what cases the above strategy would fail but this would work (since presumably if you can't exchange an AAD access token with the above API you'll also have problems using your AAD token to call this API on the management plane).  Don't think we should pull the trigger on it just yet.